src/models/Location.java

```diff
@@ -308,23 +331,19 @@ public void setY(int y) {
 
     public Guard getInvariantGuard() {
         if (invariantGuard == null) {
-            invariantGuard = getInvariantCdd().getGuard();
+            invariantGuard = getInvariantCddEager().getGuard();
         }
 
         return invariantGuard;
     }
 
-    public CDD getInvariantCdd() {
-        if (invariantGuard == null) {
-            return getInvariantCddNew();
-        }
-
+    public CDD getInvariantCddEager() {
```

Comment:
In English, eager means to strongly wanting to do or to have something. How does that apply in the context of this method?

As long as we have the `Guard` class, I would go for a method name like `getInvariantCddFromInvariantGuard()`. It might be lenghtier, but it describes more precisely what it does.

Reply:
It was a reference to eager evaluation, where an expression is evaluated when it is encountered. In this case a new instance is always created. I have removed the two different lazy/eager functions with a single one.